app/controllers/catalog_admin/choice_sets_controller.rb

```diff
@@ -115,6 +115,19 @@ def import_choice_set
     end
   end
 
+  def new_choice_modal
+    find_choice_set
+    render json: {
+      catalog: @choice_set.catalog.id, choice_set: @choice_set.id,
```

Comment:
sur 2 lignes stp, et aussi évite de charger les données des relations pour juste récupérer un id:
```suggestion
      catalog: @choice_set.catalog_id, 
      choice_set: @choice_set.id,
```